ironfish/src/wallet/account.ts

```diff
@@ -84,8 +84,10 @@ export class Account {
     }
   }
 
-  async *getNotes(): AsyncGenerator<DecryptedNoteValue & { hash: Buffer }> {
-    for await (const decryptedNote of this.walletDb.loadDecryptedNotes(this)) {
+  async *getNotes(
+    keyRange?: DatabaseKeyRange,
+  ): AsyncGenerator<DecryptedNoteValue & { hash: Buffer }> {
+    for await (const decryptedNote of this.walletDb.loadDecryptedNotes(this, keyRange)) {
```

Comment:
Im a little confused here, walletDb.loadDecryptedNotes() takes IDatabaseTransaction, how does DatabaseKeyRange converts to IDatabaseTransaction?

Reply:
After #3873 it takes an optional key range as well as an optional transaction